aerospike/src/main/java/com/yahoo/ycsb/db/AerospikeClient.java

```diff
@@ -0,0 +1,207 @@
+  private int writeOverloadTries = WRITE_OVERLOAD_TRIES;
+
+  private Policy readPolicy = new Policy();
+  private WritePolicy insertPolicy = new WritePolicy();
+  private WritePolicy updatePolicy = new WritePolicy();
+  private WritePolicy deletePolicy = new WritePolicy();
+
+  @Override
+  public void init() throws DBException {
+    insertPolicy.recordExistsAction = RecordExistsAction.CREATE_ONLY;
+    updatePolicy.recordExistsAction = RecordExistsAction.UPDATE_ONLY;
+
+    Properties props = getProperties();
+
+    namespace = props.getProperty("as.namespace", DEFAULT_NAMESPACE);
+
+    String host = props.getProperty("as.host", DEFAULT_HOST);
+    String user = props.getProperty("as.user");
+    String password = props.getProperty("as.password");
+    int port = Integer.parseInt(props.getProperty("as.port", DEFAULT_PORT));
+    int timeout = Integer.parseInt(props.getProperty("as.timeout",
+        DEFAULT_TIMEOUT));
+
+    readPolicy.timeout = timeout;
+    insertPolicy.timeout = timeout;
+    updatePolicy.timeout = timeout;
+    deletePolicy.timeout = timeout;
+
+    ClientPolicy clientPolicy = new ClientPolicy();
+
+    if (user != null && password != null) {
+      clientPolicy.user = user;
+      clientPolicy.password = password;
+    }
+
+    try {
+      client =
+          new com.aerospike.client.AerospikeClient(clientPolicy, host, port);
+    } catch (AerospikeException e) {
+      throw new DBException(String.format("Error while creating Aerospike " +
+          "client for %s:%d.", host, port), e);
+    }
+  }
+
+  @Override
+  public void cleanup() throws DBException {
+    client.close();
+  }
+
+  @Override
+  public int read(String table, String key, Set<String> fields,
+      HashMap<String, ByteIterator> result) {
+    try {
+      Record record;
+
+      if (fields != null) {
+        record = client.get(readPolicy, new Key(namespace, table, key),
+            fields.toArray(new String[fields.size()]));
+      } else {
+        record = client.get(readPolicy, new Key(namespace, table, key));
+      }
+
+      if (record == null) {
+        if (DEBUG) {
+          System.err.println("Record key " + key + " not found (read)");
+        }
+
+        return RESULT_ERROR;
+      }
+
+      for (Map.Entry<String, Object> entry: record.bins.entrySet()) {
+        result.put(entry.getKey(),
+            new ByteArrayByteIterator((byte[])entry.getValue()));
+      }
+
+      return RESULT_OK;
+    } catch (AerospikeException e) {
+      System.err.println("Error while reading key " + key + ": " + e);
+      return RESULT_ERROR;
+    }
+  }
+
+  @Override
+  public int scan(String table, String start, int count, Set<String> fields,
+      Vector<HashMap<String, ByteIterator>> result) {
+    System.err.println("Scan not implemented");
+    return RESULT_ERROR;
+  }
+
+  private int write(String table, String key, WritePolicy writePolicy,
+      HashMap<String, ByteIterator> values) {
+    if (writeOverloadTries == 0) {
```

Comment:
A little description of the retry policy here would be very helpful :).

Is the intent here that once a write fails `WRITE_OVERLOAD_TRIES` consecutive times with error code `DEVICE_OVERLOAD`, no further writes will be issued by the client?